fracatal/functional_pt/adamata.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AdamAutomaton():
```

Comment:
This is adapted from https://github.com/riveSunder/discogliders. Consider re-writing this in functional/composable format used by other CA frameworks in this project.